PilotDataPlatform/metadata-event-handler
# PILOT-1557: Create endpoint for bulk indexing metadata items into elasticsearch

## Summary

- Added FastAPI endpoint (`POST /v1/event-handler/bulk-index`) to bulk index metadata items into Elasticsearch in the event Kafka is unexpectedly offline. This endpoint will either bulk create or update existing Elasticsearch documents through relying on the metadata service to query all items last updated between a date-range (range when Kafka was offline).
- Endpoint encapsulates the event-handler (consumer).

## JIRA Issues

https://indocconsortium.atlassian.net/browse/PILOT-1557

## Type of Change


## Testing

Are there any new or updated tests to validate the changes?


## Test Directions

1. Call `POST /v1/event-handler/bulk-index` with date-range.
2. Check that the metadata items between the date-range are inserted/updated in the correct Elasticsearch index.
